### ISSUE TITLE
jOOQ Open Source Edition does not generate routine with long name

### ISSUE THREAD

**Issue by tmsklr** [jOOQ/jOOQ#12970]
### Expected behavior
JOOQ generates a routine in the metamodel for each PostgreSQL function.

### Actual behavior
JOOQ does not generate a corresponding routine if its name is longer than 63 characters. 

A slightly shorter name may also reproduce the problem. As far as I have debugged, the wrong join condition in org.jooq.meta.postgres.PostgresDatabase#getRoutines0 is used because org.jooq.meta.postgres.PostgresDatabase#is12 returns false (https://github.com/jOOQ/jOOQ/issues/11325).

### Steps to reproduce the problem
- Create a PostgreSQL function with a name of length > 63
- Run the code generator

### Versions

- jOOQ: 3.15.4
- Java: 13.0.8
- Database (include vendor): PostgreSQL 13.5
- OS: macOS 
- JDBC Driver (include name if inofficial driver): 42.3.1

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can confirm this works only with the commercial distributions (which previously check the `SQLDialect.POSTGRES_13` type to prevent having to query the `pg_catalog` for meta data view existence). As we can't use this optimisation in the Open Source Edition, the query then fails, because it's wrong.

**Comment by lukaseder** [comment]
The problem is caused by the override of `AbstractDatabase::exists0` in `PostgresDatabase`:

```java
    @Override
    protected boolean exists0(TableField<?, ?> field) {
        return exists1(field, COLUMNS, COLUMNS.TABLE_SCHEMA, COLUMNS.TABLE_NAME, COLUMNS.COLUMN_NAME);
    }
```

That override is implemented for performance reasons and to prevent logging of exceptions in some logging configurations (https://github.com/jOOQ/jOOQ/issues/8972), but it prevents querying for the existence of routines.

I'll just implement a different way to check if we're on PostgreSQL 12+

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.4 (https://github.com/jOOQ/jOOQ/issues/12971), and 3.15.8 (https://github.com/jOOQ/jOOQ/issues/12972)
